pkg/capi/cluster.go

```diff
@@ -247,3 +251,13 @@ func (cluster *Cluster) health(ctx context.Context) error {
 func (cluster *Cluster) Name() string {
 	return cluster.name
 }
+
+// Namespace of the cluster.
+func (cluster *Cluster) Namespace() string {
+	return cluster.namespace
+}
+
+// CAPIVersion that was used to deploy the cluster.
+func (cluster *Cluster) CAPIVersion() string {
```

Comment:
not for now, but we should see if we can figure out a way to determine this for the user. I guess once we go to v1alpha4, all clusters get mutated to v1alpha4 anyways on the backend.

Reply:
Would be nice to query cluster without version somehow I guess.

Reply:
autodetect works only when you deploy a cluster though :) If we want to run destroy for already deployed cluster, it's going to be more tricky.